sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/utils/CalciteUtils.java

```diff
@@ -109,7 +99,7 @@ public static boolean isStringType(FieldType fieldType) {
   public static final FieldType BOOLEAN = FieldType.BOOLEAN;
   public static final FieldType VARBINARY = FieldType.BYTES;
   public static final FieldType VARCHAR = FieldType.STRING;
-  public static final FieldType CHAR = FieldType.logicalType(new CharType());
+  public static final FieldType CHAR = FieldType.STRING;
```

Comment:
To avoid breaking change this public field is preserved but is now the same as VARCHAR (originally Calcite.CharType was also essentially the same as VARCHR. It did not do padding as FixedString.)

Reply:
I disagree. CHAR is not the same as string, it should have fixed length. What breaks if you make this fixed length?

Reply:
Yeah, I agree. The "breaking change" I mentioned here is the removal of a public field.

We have the same situation for fixed binary:
https://github.com/apache/beam/blob/60e465e40d413ea6fb8c113ff5619c30be5864b0/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/utils/CalciteUtils.java#L147
and
https://github.com/apache/beam/blob/60e465e40d413ea6fb8c113ff5619c30be5864b0/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/utils/CalciteUtils.java#L100
makes BINARY the same as BYTES in CalciteUtils currently.

A full support for BINARY/VARBINARY/CHAR/VARCHAR with specified limit involves taking Calcite logical type's precision parameter (currently it is omitted) and pass it to FixedBinary/VariableBinary/FixedString/VariableString constructors. We cannot have constant logical type instance for these types because they require an argument in constructor. The full support for these types is left as future work. For now this change just unblock the use case that mentioned in https://issues.apache.org/jira/browse/BEAM-12323.

Reply:
Entered #24019 and added TODO here. Thanks!

Reply:
Hi @apilloud, do you think it suffices to map CHAR to String as we did for mapping BINARY to bytes, or we should also implement #24019 in this PR. Is there any breaking change if treat CHAR as String in Beam?

Reply:
I believe that this path is only used by string literals used in a SQL statement today. Assuming that is true there is no harm in moving forward with this as is. I wanted to know what breaks here as that would potentially identify other use cases.